manual/root_io/index.md

```diff
@@ -1,348 +1,222 @@
-#include "TObject.h"
-
-class TEvent;
-class TTrack : public TObject {
-private:
-   Int_t fId;       // Track sequential id.
-   TEvent *fEvent;  // TEvent to which track belongs.
-   Float_t fPx;     // x part of track momentum.
-   Float_t fPy;     // y part of track momentum.
-   Float_t fPz;     // z part of track momentum.
-public:
-   TTrack() { fId = 0; fEvent = 0; fPx = fPy = fPz = 0; }
-   TTrack(Int_t id, Event *ev, Float_t px,Float_t py,Float_t pz);
-   Float_t Momentum() const;
-   TEvent *GetEvent() const { return fEvent; }
-   void Print(Option_t *opt="");
-   ClassDef (TTrack,1);    //Simple track class.
-};
-{% endhighlight %}
+- [using ACLiC](#using-aclic): use this method to generate class dictionaries for quick prototyping and interactive development.
+- [using `rootcling`](#using-rootcling): this is a low level command line tool to generate dictionaries. You can invoke `rootcling` e.g. from a Makefile or a shell script.
+- [using CMake](#using-cmake): use this method to integrate ROOT I/O in your C++ framework build system.
 
-**Implementation of TEvent and TTrack class**
+### Using ACLiC
 
-{% highlight C++ %}
-TEvent.cxx:
-#include <iostream.h>
-#include "TOrdCollection.h"
-#include "TEvent.h"
-#include "TTrack.h"
-...
-
-TTrack.cxx:
-#include <iostream.h>
-#include "TMath.h"
-#include "Track.h"
-#include "Event.h"
-...
-{% endhighlight %}
+When you compile code from the ROOT prompt using [ACLiC]({{ '/manual/root_macros_and_shared_libraries/#compiling-root-macros-with-aclic' | relative_url }}), ROOT automatically generates the dictionaries for the types defined in that code and compiles them into a shared library.
 
-**Using rootcling to generate the dictionary**
+For a standalone source file `MyClass.cxx`, we can interactively compile the source file into a library and, at the same time, create dictionaries for the types defined in it, with:
 
 {% highlight C++ %}
-rootcling eventdict.cxx -c TEvent.h TTrack.h
+root[] .L MyClass.cxx+
 {% endhighlight %}
 
-**eventdict.cxx - the generated dictionary**
+At this point, the ROOT interpreter has loaded all the information required to perform I/O of the types in `MyClass.cxx`:
 
 {% highlight C++ %}
-void TEvent::Streamer(TBuffer &R__b) {
-
-// Stream an object of class TEvent.
-   if (R__b.IsReading()) {
-      Version_t R__v = R__b.ReadVersion();
-      TObject::(R__b);
-      R__b >> fId;
-      R__b >> fTotalMom;
-      R__b >> fTracks;
-   } else {
-      R__b.WriteVersion(TEvent::IsA());
-      TObject::Streamer(R__b);
-      R__b << fId;
-      R__b << fTotalMom;
-      R__b << fTracks;
-   }
-}
+MyClass c;
+TFile f("myfile.root", "RECREATE"); // create a file called `f.root`
+f.WriteObjectAny(&c, "MyClass", "c"); // write object `c` as key `c` into the file
 {% endhighlight %}
 
-## Selecting dictionary entries: Linkdef.h
+The compiled dictionary will be called `MyClass_cxx.so` (and, by default, the generated dictionary source file is automatically deleted).
```

Comment:
Consider
```
The library containing the compiled dictionary will be called `MyClass_cxx.so` (
```

Reply:
done in https://github.com/root-project/web/pull/735